patch-diff-report-tool/src/main/java/com/github/checkstyle/parser/EmptyXmlEventReader.java

```diff
@@ -0,0 +1,71 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.github.checkstyle.parser;
+
+import java.util.NoSuchElementException;
+
+import javax.xml.stream.XMLEventReader;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.events.XMLEvent;
+
+/**
+ * Simple {@link XMLEventReader} with no elements.
+ * @author Richard Veach
+ */
+public final class EmptyXmlEventReader implements XMLEventReader {
```

Comment:
> Do we need to introduce EmptyXmlEventReader to keep logic unchanged in parse ? 

That was my main reasoning, yes.
We still need a change in `StaxUtils.createReader` because it will get a NPE. We also pass the reader around, so who knows if we would have to make more places null safe.